### ISSUE TITLE
Ghost introductions / delete function

### ISSUE THREAD

**Issue by beardyunixer** [friendica/friendica#3409]
It is possible to find a ghost introduction from an unknown user at network unkn.

These happen when somebody from ostatus (I suspect, but haven't tested, if this also occurs from Diaspora) follows you, then unfollows you before you've seen the introduction.  

The problem is lose_follower() and lose_sharer() both delete the contact (via contact_remove() where the actual deletion takes place), but do not check to see if there are any pending introductions from this contact before doing so, and so the user is left with an introduction without a corresponding contact record.

Michael believes we need a new delete function to deal with this kind of deletion.

**Comment by annando** [comment]
Currently I'm enhancing ```dbstructure.php``` so that it documents the table relations as well. This will be the first step.

**Comment by annando** [comment]
See this pull request: https://github.com/friendica/friendica/pull/3415

We now have to use this function at all places where we delete data.

**Comment by annando** [comment]
@beardyunixer can you try again?

**Comment by Shnoulle** [comment]
Hi, seems i have same issue :). Currently i'm on last master (of this morning). I do a /admin/dbsync/check and i can't manually delete item since i didn't know the GUID.

For information, i update this morning, but i think previous version are newer than may ?

**Comment by annando** [comment]
From my point of view this already should be solved.

**Comment by tobiasd** [comment]
So we can close this issue, or was somebody visited by ghosts recently?

**Comment by AlfredSK** [comment]
Hmm. I'm not sure. The thing I discovered maybe related or not...

Introductions from OStatus are often (always?) labeled with [Name Withheld] in the notification mail about the new contact. I don't know whether this is on purpose or a bug.

But what is really confusing when I click the notification in Friendica I'm directed to a completely empty¹  .../notifications/intros page. I have to click on a link labeled "show all..." or something like that to see the new introduction.

1 - with "empty" I _don't_ refer to a WSOD

**Comment by annando** [comment]
@AlfredSK This empty page could be some adblocker issue.

@tobiasd I think that we can close this issue. The other issue not related, I think.

**Comment by AlfredSK** [comment]
> This empty page could be some adblocker issue.

I don't think so. :) But you can close this. I think the behavior I discovered is unrelated.
If I see my issue again I will open a separate bug report.
